cmake/UseCcache.cmake

```diff
@@ -0,0 +1,56 @@
+# Compiling ccache via ccache is fine, because this uses a stable version which
+# is installed on the system.
+
+# Modified version of Craig Scott's "Professional CMake: A Practical Guide", 7th Edition
+function(use_ccache)
+    if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
```

Comment:
Since we're indenting CMake scripts with two spaces elsewhere I think that it would make sense to do so here as well even though it's based on an example that uses four space indentation.